sdk/tests/deqp/framework/opengl/gluDrawUtil.js

```diff
@@ -40,7 +40,7 @@ var gluShaderProgram = framework.opengl.gluShaderProgram;
  */
 gluDrawUtil.VertexArrayBinding = function(type, location, components, elements, data, stride, offset) {
     this.type = type;
-    this.location = location === undefined ? -1 : location;
+    this.location = location === undefined ? 0 : location;
```

Comment:
It looks to me like this maps to the "location" field in the native dEQP's BindingPoint class:
https://android.googlesource.com/platform/external/deqp/+/master/framework/opengl/gluDrawUtil.hpp#105

and that (given the change below to the addition of "location"), setting it to -1 was causing the wrong value to be added / computed for attributes' locations in these tests.

Qiankun, is that correct?